google-cloud-clients/google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/it/ReadIT.java

```diff
@@ -111,6 +117,31 @@ public void read() throws Throwable {
     assertThat(observer.responses).containsExactlyElementsIn(expectedRows);
   }
 
+  @Test
+  public void readSingleNonexistentAsyncCallback() throws Exception {
+    ApiFuture<Row> future = testEnvRule.env().getDataClient()
+        .readRowAsync(testEnvRule.env().getTableName().getTable(), "somenonexistentkey");
+
+    final AtomicBoolean found = new AtomicBoolean();
+    final CountDownLatch latch = new CountDownLatch(1);
```

Comment:
why do you need both a `CountDownLatch` and an `AtomicBoolean`? Can you combine them or use `future.get()`?

Reply:
I was trying to reproduce this issue:
https://github.com/googleapis/google-cloud-java/pull/3856#discussion_r228957914